docs/changes/cosmology/12176.api.rst

```diff
@@ -0,0 +1 @@
+The function ``astropy.cosmology.utils.vectorize_if_needed()`` is removed.
```

Comment:
I don't think we need to mention removal of private functions (if we do, there should be a deprecation...)

Reply:
In v4.3- this function lived directly in cosmology/core.py along with all the publicly scoped functions. Only recently was it moved to utils.py to make it more clear that this is not publicly scoped.
Additionally it isn't prefixed with an underscore, so I think it's easily mistaken for public API. I just feel more comfortable with the changelog. When I remove  `_float_or_None`no changelog will be needed.

Reply:
OK, makes sense. In that case, it may make most sense to also properly deprecate... (`astropy.utils.deprecated`)

Reply:
Done! I'm going to do this to ``inf_like`` as well.